app-react/src/main/kotlin/com/ustadmobile/view/HolidayCalendarEditScreen.kt

```diff
@@ -0,0 +1,123 @@
+package com.ustadmobile.view
+
+import com.ustadmobile.core.generated.locale.MessageID
+import com.ustadmobile.core.hooks.useStringsXml
+import com.ustadmobile.core.viewmodel.HolidayCalendarEditUiState
+import com.ustadmobile.hooks.useFormattedDate
+import com.ustadmobile.lib.db.entities.Holiday
+import com.ustadmobile.lib.db.entities.HolidayCalendar
+import com.ustadmobile.lib.db.entities.ext.shallowCopy
+import com.ustadmobile.mui.components.UstadTextEditField
+import kotlinx.css.px
+import mui.icons.material.Add
+import mui.icons.material.Delete
+import mui.material.*
+import mui.material.styles.TypographyVariant
+import mui.system.responsive
+import react.FC
+import react.Props
+import react.ReactNode
+import react.create
+
+external interface HolidayCalendarEditProps: Props {
+    var uiState: HolidayCalendarEditUiState
+    var onAddItemClick: () -> Unit
+    var onHolidayCalendarChange: (HolidayCalendar?) -> Unit
+    var onDeleteItemClick: (Holiday?) -> Unit
+    var onItemClick: (Holiday) -> Unit
+}
+
+val HolidayCalendarEditComponent2 = FC<HolidayCalendarEditProps> { props ->
+
+    val strings = useStringsXml()
+
+    Container{
+        maxWidth = "lg"
+
+        Stack{
+            direction = responsive(mui.material.StackDirection.column)
+            spacing = responsive(10.px)
+
+            UstadTextEditField {
+                value = props.uiState.holidayCalendar?.umCalendarName ?: ""
+                label = strings[MessageID.name]
+                enabled = props.uiState.fieldsEnabled
+                onChange = {
+                    props.onHolidayCalendarChange(
+                        props.uiState.holidayCalendar?.shallowCopy {
+                            umCalendarName = it
+                        })
+                }
+            }
+
+            ListItem {
+
+                disablePadding = true
+
+                ListItemButton{
+                    ListItemIcon {
+                        Add{}
+                    }
+
+                    ListItemText{
+                        primary = ReactNode((strings[MessageID.add_a_holiday]))
+                    }
+
+                    onClick = {
+                        props.onAddItemClick()
+                    }
+                }
+            }
+
+            List{
+                props.uiState.holidayList?.forEach { holiday ->
+                    ListItem {
```

Comment:
The holiday itself does not have an icon, but you need to use a blank icon so that the spacing will be correct and the text will line up. The text for "add a holiday" and "Eid" should align, 

![spacer](https://user-images.githubusercontent.com/4280622/209865796-28adcf99-f8cd-47ac-a80e-5cc03aa9c511.png)

The way to do this is to add a blank icon so that the text will start at the same position.

e.g. add this to the listitem.

```
ListItemIcon {
   UstadBlankIcon { } 
}
```